packages/expo-modules-core/ios/AppDelegates/ExpoAppDelegate.swift

```diff
@@ -37,7 +35,16 @@ open class ExpoAppDelegate: UIResponder, UIApplicationDelegate {
     }
   }
 
-  open func application(_ application: UIApplication, didFinishLaunchingWithOptions launchOptions: [UIApplication.LaunchOptionsKey: Any]? = nil) -> Bool {
+  open override func application(
+    _ application: UIApplication,
+    didFinishLaunchingWithOptions launchOptions: [UIApplication.LaunchOptionsKey: Any]? = nil
+  ) -> Bool {
+    if UIApplication.shared.delegate is ExpoAppDelegate {
+      // Super function from `RCTAppDelegate` must be called only when the class is used as the base class of the AppDelegate.
+      // For backwards compatibility `EXAppDelegateWrapper` creates a detached instance of `ExpoAppDelegate`, in which case this call must be skipped.
+      super.application(application, didFinishLaunchingWithOptions: launchOptions)
+    }
+
     return subscribers.reduce(false) { result, subscriber in
       return subscriber.application?(application, didFinishLaunchingWithOptions: launchOptions) ?? false || result
```

Comment:
the `didFinishLaunchingWithOptions` returns true previously. now the pr changes the final return value through the subscribers and returns false. it could be a regression also.

Reply:
Yup, @lukmccall already caught this too, we discussed that yesterday and we think it's best to just change it to `forEach` instead of `reduce`, see https://github.com/expo/expo/pull/33348#discussion_r1868040497